src/VaultRouter.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.22;
+
+import {SafeERC20, IERC20} from "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import {IVaultRouter, IVault} from "./IVaultRouter.sol";
+import {
+    AccessControlDefaultAdminRules,
+    AccessControl,
+    IAccessControl
+} from "openzeppelin-contracts/contracts/access/extensions/AccessControlDefaultAdminRules.sol";
+
+/// @title VaultRouter Contract
+/// @notice This contract acts as a router to interact with the Vault for fund withdrawals and deposits.
+contract VaultRouter is IVaultRouter, AccessControlDefaultAdminRules {
```

Comment:
This router isn't needed in it's current form. Let's solve the issue of bundling vault and order calls in a separate PR. The best solution I can think of - using smart contract wallets for our fulfillers - wouldn't need a special contract in this codebase.